thunder/core/module.py

```diff
@@ -141,6 +143,116 @@ def load_original_state_dict(self, state_dict):
+        for name, submodule in self._model.named_modules(prefix=prefix):
+            subprefix = f"{name}." if name else name
+            non_persistent_buffers_set.update(f"{subprefix}{bname}" for bname in submodule._non_persistent_buffers_set)
+
+        for name, buf in self.named_buffers(prefix=prefix):
+            if buf is not None and name not in non_persistent_buffers_set:
+                destination[name] = buf if keep_vars else buf.detach()
+
+        for name, submodule in self._model.named_modules(prefix=prefix):
+            subprefix = f"{name}." if name else name
+            extra_state_key = subprefix + pytorch.nn.modules.module._EXTRA_STATE_KEY_SUFFIX
+            if (
+                getattr(self.__class__, "get_extra_state", pytorch.nn.Module.get_extra_state)
+                is not pytorch.nn.Module.get_extra_state
+            ):
+                destination[extra_state_key] = self.get_extra_state()
+        return destination
+
+    def load_state_dict(self, state_dict: Mapping[str, Any], strict: bool = True, assign: bool = False):
+        """Loads the state dict to a transformed module.
+
+        Args:
+            state_dict: the state dict to load.
+            strict: error on missing / unused state dict members
+            assign: assign the state dict tensors instead of copying the data
+
+        This is similar much more simple than the original load_state_dict.
+        (Regarding hooks, customization etc.)
+        """
+        # non-persistent buffer overrides?
+        non_persistent_buffers_set = set()
+        for name, submodule in self._model.named_modules():
+            subprefix = f"{name}." if name else name
+            non_persistent_buffers_set.update(f"{subprefix}{bname}" for bname in submodule._non_persistent_buffers_set)
+
+        keys_unused = thunder.core.utils.OrderedSet(state_dict.keys())
+        keys_missing = thunder.core.utils.OrderedSet()
+        errors = []
+        for name, v in itertools.chain(self.named_parameters(), self.named_buffers()):
+            if name in non_persistent_buffers_set:
+                continue
+            if name not in state_dict:
+                keys_missing.add(name)
+                continue
+            keys_unused.remove(name)
+            sd_v = state_dict[name]
+            if not isinstance(sd_v, pytorch.Tensor):
+                errors.append(
+                    f'While copying the parameter named "{name}", '
+                    "expected torch.Tensor or Tensor-like object from checkpoint but "
+                    f"received {type(sd_v)}"
+                )
+                continue
+            if sd_v.shape != v.shape:
+                errors.append(
+                    f"size mismatch for {name}: copying a param with shape {sd_v.shape} from checkpoint, "
+                    f"the shape in current model is {v.shape}."
+                )
+                continue
+            # check dtype?
```

Comment:
is this a TODO or a NOTE?

Reply:
Probably a note that we don't because PyTorch does not.